runtime/lua/vim/treesitter/query.lua

```diff
@@ -262,44 +262,62 @@ function M.get_node_text(...)
   return vim.treesitter.get_node_text(...)
 end
 
----@alias TSMatch table<integer,TSNode>
+---@alias TSMatch table<integer,TSNode[]>
 
----@alias TSPredicate fun(match: TSMatch, _, _, predicate: any[]): boolean
+---@alias TSPredicate fun(match: TSMatch, _, _, predicate: any[], any: boolean): boolean
 
 -- Predicate handler receive the following arguments
--- (match, pattern, bufnr, predicate)
+-- (match, pattern, bufnr, predicate, any)
 ---@type table<string,TSPredicate>
 local predicate_handlers = {
-  ['eq?'] = function(match, _, source, predicate)
-    local node = match[predicate[2]]
-    if not node then
+  ['eq?'] = function(match, _, source, predicate, any)
+    local nodes = match[predicate[2]]
+    if not nodes or #nodes == 0 then
       return true
     end
-    local node_text = vim.treesitter.get_node_text(node, source)
 
-    local str ---@type string
-    if type(predicate[3]) == 'string' then
-      -- (#eq? @aa "foo")
-      str = predicate[3]
-    else
-      -- (#eq? @aa @bb)
-      str = vim.treesitter.get_node_text(match[predicate[3]], source)
-    end
+    for _, node in ipairs(nodes) do
+      local node_text = vim.treesitter.get_node_text(node, source)
 
-    if node_text ~= str or str == nil then
-      return false
+      local str ---@type string
+      if type(predicate[3]) == 'string' then
+        -- (#eq? @aa "foo")
+        str = predicate[3]
+      else
+        -- (#eq? @aa @bb)
+        local other = assert(match[predicate[3]])
+        assert(#other == 1, '#eq? does not support matching against multiple captures')
```

Comment:
The semantics of `#eq?` should be: "all" the capture nodes equal to `rhs`. This is the semantic the upstream tree-sitter is using, as opposed to `#any-eq?`.

https://tree-sitter.github.io/tree-sitter/using-parsers#pattern-matching-with-queries

Reply:
Yes, and that is what I have implemented. This line in particular handles the case when `rhs` is another capture, and that other capture contains multiple nodes. It's not at all clear what the behavior in that situation should be, so for now we just don't allow it. We can always define that behavior later and remove this assertion.

Reply:
Oh I see, I'm sorry I got this wrong. Yes, it looks undefined in the upstream tree-sitter at the moment.

Reply:
You're right though that the error message is incorrect. It should say "matching against a capture containing multiple nodes".